### ISSUE TITLE
JSON parser example fails on escaped strings and empty arrays

### ISSUE THREAD

**Issue by nrkn** [dominictarr/stack-expression#3]
```json
{
  "test": "echo \"Error: no test specified\" && exit 1"
}
```

```json
{
  "keywords": []
}
```

**Comment by nrkn** [comment]
Additionally, it currently allows fractions in `e` notation:

```js
var decimal = AND(int, MAYBE(fraction))
var number = TEXT(AND(decimal, MAYBE(AND('e', decimal))), Number)
```

```js
console.log(
  json('1e1.1',0).groups[0]
)
```

```
NaN
```

**Comment by dominictarr** [comment]
oh, those should be non-zero integers

**Comment by dominictarr** [comment]
any integer except zero

**Comment by dominictarr** [comment]
string escapes fixed in: f3f2d276603cebca4a2f45bdbd67330f201e3818
the parser for non-escaped chars needed to reject the escape character.
f3f2d276603cebca4a2f45bdbd67330f201e3818

if it had just accepted a single character at a time it would have been fine, but it's way faster with a greedy regexp `/^...+/`
